src/compiler/checker.ts

```diff
@@ -8223,6 +8459,60 @@ module ts {
             }
         }
 
+        function getPromiseConstructor(node: SignatureDeclaration): EntityName {
+            if (node.type && node.type.kind === SyntaxKind.TypeReference) {
+                var typeReference = <TypeReferenceNode>node.type;
+                return typeReference.typeName;
+            }
+
+            if (globalPromiseSymbol && globalPromiseSymbol === resolveName(node, "Promise", SymbolFlags.Value, undefined, undefined)) {

```

Comment:
Actually, I guess all the locals will be inside the generator body that you emit. But what about parameters named Promise? Would those be emitted on the outer function or the generator?